internal/api/inferenceservice/infer.go

```diff
@@ -51,12 +49,20 @@ func doInfer(ctx context.Context, rebuilder rebuild.Rebuilder, t rebuild.Target,
 type InferDeps struct {
 	HTTPClient httpx.BasicClient
 	GitCache   *gitx.Cache
+	Storage    storage.Storer
```

Comment:
nit: so i think we should model these Deps structs as per-server, not per-instance. in this case, i think we'd want this to be a callback that returns the repo options. that way, each request would receive a different instance of the pair (even though the current local stub impl could just hard-code a single storer and fs for each request).